nilearn/tests/test_signal.py

```diff
@@ -1075,8 +1076,79 @@ def test_handle_scrubbed_volumes():
         interpolated_confounds[sample_mask, :], confounds[sample_mask, :]
     )
 
-    scrubbed_signals, scrubbed_confounds = nisignal._handle_scrubbed_volumes(
-        signals, confounds, sample_mask, "cosine", 2.5
+    (
+        scrubbed_signals,
+        scrubbed_confounds,
+        sample_mask,
+    ) = nisignal._handle_scrubbed_volumes(
+        signals, confounds, sample_mask, "cosine", 2.5, True
     )
     np.testing.assert_equal(scrubbed_signals, signals[sample_mask, :])
     np.testing.assert_equal(scrubbed_confounds, confounds[sample_mask, :])
+
+
+def test_handle_scrubbed_volumes_extrapolation():
```

Comment:
@jhuguetn this test has become a bit long. I suggest splitting it up even if it means repeating some data generation lines. Maybe two separate tests one with extrapolation and the other without will do

Reply:
Agree, it makes sense to divide the content into 2 tests for a better code readability. See bd8e627.